sensor/debugger/k8s/fake.go

```diff
@@ -436,6 +437,123 @@ func (f *FakeEventsManager) waitOnMode(events []string) error {
 	return nil
 }
 
+// forEachResource executes a given function to all resources of a given kind
+func (f *FakeEventsManager) forEachResource(client reflect.Value, execFunc func(reflect.Value) error) error {
+	returnVals := client.MethodByName("List").Call([]reflect.Value{
+		reflect.ValueOf(context.Background()),
+		reflect.ValueOf(metav1.ListOptions{}),
+	})
+
+	if len(returnVals) != 2 {
+		return fmt.Errorf("expected 2 values from %s. Received: %d", "List", len(returnVals))
+	}
+	errInt := returnVals[len(returnVals)-1].Interface()
+	if errInt != nil {
+		return errInt.(error)
+	}
+
+	errorList := errorhelpers.NewErrorList("for each resource")
+	items := reflect.Indirect(returnVals[0]).FieldByName("Items")
+	if !items.IsValid() {
+		errorList.AddError(errors.New("invalid kind: missing `Items` property"))
+	} else {
+		for i := 0; i < items.Len(); i++ {
+			resource := items.Index(i)
+			if err := execFunc(resource); err != nil {
+				errorList.AddError(err)
+			}
+		}
+	}
+	return errorList.ToError()
+}
+
+// getNameFromObjectMeta returns the name of the resource
+func getNameFromObjectMeta(obj reflect.Value) (string, error) {
+	objMeta := obj.FieldByName("ObjectMeta")
+	if !objMeta.IsValid() {
+		return "", errors.New("resource does not have the ObjectMeta field")
+	}
+	objMetaPtr := objMeta.Addr()
```

Comment:
`objMetaPtr` is used only in the next line - maybe a candidate to remove?

Reply:
Done